core/scripts/server.js

```diff
@@ -208,35 +204,52 @@ app.get('*', (req, res, next) => {
     }).catch(errorHandler)
   }
 
-  if (config.server.useOutputCache && cache) {
-    cache.get(
-      'page:' + req.url
-    ).then(output => {
-      if (output !== null) {
-        if (output.headers) {
-          for (const header of Object.keys(output.headers)) {
-            res.setHeader(header, output.headers[header])
+  const dynamicCacheHandler = () => {
+    if (config.server.useOutputCache && cache) {
+      cache.get(
+        'page:' + req.url
+      ).then(output => {
+        if (output !== null) {
+          if (output.headers) {
+            for (const header of Object.keys(output.headers)) {
+              res.setHeader(header, output.headers[header])
+            }
           }
-        }
-        res.setHeader('X-VS-Cache', 'Hit')
-        if (output.body) {
-          res.end(output.body)
+          res.setHeader('X-VS-Cache', 'Hit')
+          if (output.body) {
+            res.end(output.body)
+          } else {
+            res.setHeader('Content-Type', 'text/html')
+            res.end(output.body)
+          }
+          res.end(output)
+          console.log(`cache hit [${req.url}], cached request: ${Date.now() - s}ms`)
+          next()
         } else {
           res.setHeader('Content-Type', 'text/html')
-          res.end(output.body)
+          res.setHeader('X-VS-Cache', 'Miss')
+          console.log(`cache miss [${req.url}], request: ${Date.now() - s}ms`)
+          dynamicRequestHandler(renderer) // render response
         }
-        res.end(output)
-        console.log(`cache hit [${req.url}], cached request: ${Date.now() - s}ms`)
-        next()
-      } else {
-        res.setHeader('Content-Type', 'text/html')
-        res.setHeader('X-VS-Cache', 'Miss')
-        console.log(`cache miss [${req.url}], request: ${Date.now() - s}ms`)
-        dynamicRequestHandler(renderer) // render response
-      }
-    }).catch(errorHandler)
+      }).catch(errorHandler)
+    } else {
+      dynamicRequestHandler(renderer)
+    }
+  }
+
+  if (config.server.dynamicConfigReload) {
+    delete require.cache[require.resolve('config')]
+    if (typeof serverExtensions.loadConfig === 'function') {
```

Comment:
Can we remove “loadConfig” to “configProvider”? I would make it a convention of providers for other resources as well

Reply:
Sure, this is a great idea. Will do.